cfgmgr/buffermgrdyn.cpp

```diff
@@ -60,13 +60,7 @@ BufferMgrDynamic::BufferMgrDynamic(DBConnector *cfgDb, DBConnector *stateDb, DBC
     if (nullptr != zeroProfilesInfo)
         m_zeroPoolAndProfileInfo = *zeroProfilesInfo;
 
-    string platform = getenv("ASIC_VENDOR") ? getenv("ASIC_VENDOR") : "";
-    if (platform == "")
-    {
-        SWSS_LOG_ERROR("Platform environment variable is not defined, buffermgrd won't start");
-        return;
-    }
-
+    string platform = getenv("ASIC_VENDOR") ? getenv("ASIC_VENDOR") : "mock_test";
```

Comment:
Not in favor of checking mock_test in platform string in the code. @neethajohn , what do you think?

Reply:
We need a way to indicate whether it is mock test or real world scenario.
Any other suggestions?

Reply:
so currently for mock_test this field is "empty"?

Reply:
ok, @neethajohn , can you suggest a way to initialize this without hard-coding "mock_test". AFAIK, we don't have this elsewhere and don't prefer to introduce it here.

Reply:
I updated it. Now `mock_test` is set in the UT code. this part won't be changed.
@prsunny